src/persistence/job.js

```diff
@@ -12,18 +12,58 @@ module.exports = {
       }, ${jobParameters.description}, current_date, ${jobParameters.tags}, ${
         jobParameters.company
       }, ${jobParameters.logoURL})
-        RETURNING id, title, tags;
+        RETURNING *;
       `);
       return rows[0];
     } catch (error) {
       throw error;
     }
   },
-
   async find(id) {
     const {rows} = await db.query(sql`
     SELECT * FROM jobs WHERE id=${id} LIMIT 1;
     `);
     return rows[0];
+  },
+  async update(job, jobParameters) {
+    try {
+      let newTitle = jobParameters.title.trim();
+      let newSalaryRange = jobParameters.salaryRange.trim();
+      let newDescription = jobParameters.description.trim();
+      let newTags = jobParameters.tags;
+      let newCompany = jobParameters.company.trim();
+      let newLogoURL = jobParameters.logoURL.trim();
+      if (newTitle === '') {
+        newTitle = job.title;
+      }
+
+      if (newSalaryRange.trim() === '') {
+        newSalaryRange = job.salaryRange;
+      }
+
+      if (newDescription.trim() === '') {
+        newDescription = job.description;
+      }
+
+      if (newTags.length === 0) {
+        newTags = job.tags;
+      }
+
+      if (newCompany.trim() === '') {
+        newCompany = job.company;
+      }
+
+      if (newLogoURL.trim() === '') {
+        newLogoURL = job.logoURL;
+      }
+
+      const {rows} = await db.query(sql`
+        UPDATE jobs SET title = ${newTitle}, salary_range = ${newSalaryRange}, description = ${newDescription}, tags = ${newTags}, company = ${newCompany}, logo_url = ${newLogoURL} WHERE id = ${job.id}
+          RETURNING id, title, salary_range, description, create_at, tags, company, logo_url ;
+        `);
+      return rows[0];
+    } catch (error) {
+      throw error;
```

Comment:
It doesn't make sense to catch and throw an error immediately within a function.